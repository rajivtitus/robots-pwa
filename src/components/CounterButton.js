import React, { Component } from "react";

class CounterButton extends Component {
	constructor(props) {
		super(props);
		this.state = {
			count: 0,
		};
	}

	shouldComponentUpdate(nextProps, nextState) {
		if (this.state.count !== nextState.count) {
			return true;
		}
		return false;
	}

	handleChange = () => {
		this.setState((state) => ({ count: state.count + 1 }));
	};

	render() {
		return (
			<button id="counter" color={this.props.color} onClick={this.handleChange}>
				Count: {this.state.count}
			</button>
		);
	}
}

export default CounterButton;
