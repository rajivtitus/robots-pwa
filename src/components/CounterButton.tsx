import React, { Component } from "react";

interface Props {}

interface State {
	count: number;
}

class CounterButton extends Component<Props, State> {
	constructor(props: Props) {
		super(props);
		this.state = {
			count: 0,
		};
	}

	shouldComponentUpdate(nextProps: Props, nextState: State) {
		if (this.state.count !== nextState.count) {
			return true;
		}
		return false;
	}

	handleChange = () => {
		this.setState((state) => ({ count: state.count + 1 }));
	};

	render() {
		return (
			<button id="counter" onClick={this.handleChange}>
				Count: {this.state.count}
			</button>
		);
	}
}

export default CounterButton;
