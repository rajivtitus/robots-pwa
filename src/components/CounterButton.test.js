import { shallow } from "enzyme";

import CounterButton from "./CounterButton";

const mockColor = "red";

it("renders the counterbutton commponent", () => {
	expect(shallow(<CounterButton color={mockColor} />)).toMatchSnapshot();
});

it("increments the CounterButton commponent", () => {
	const wrapper = shallow(<CounterButton color={mockColor} />);
	expect(wrapper).toMatchSnapshot();

	expect(wrapper.props().color).toEqual("red");
	wrapper.find('[id="counter"]').simulate("click");
	expect(wrapper.state()).toEqual({ count: 1 });
});
