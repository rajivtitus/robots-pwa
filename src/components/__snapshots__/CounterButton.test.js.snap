// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`increments the CounterButton commponent 1`] = `
<button
  color="red"
  id="counter"
  onClick={[Function]}
>
  Count: 
  0
</button>
`;

exports[`renders the counterbutton commponent 1`] = `
<button
  color="red"
  id="counter"
  onClick={[Function]}
>
  Count: 
  0
</button>
`;
